fgpyo/fasta/builder.py

```diff
@@ -0,0 +1,231 @@
+"""
+Classes for generating fasta files and records for testing
+----------------------------------------------------------
+This module contains utility classes for creating fasta files, indexed fasta files (.fai), and
+sequence dictionaries (.dict).
+
+Examples of creating sets of contigs for writing to fasta
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Writing a FASTA with two contigs each with 100 bases.
+.. code-block:: python
+    >>> from fgpyo.fasta.builder import FastaBuilder
+    >>> builder = FastaBuilder()
+    >>> builder.add("chr10").add("AAAAAAAAAA", 10)
+    >>> builder.add("chr11").add("GGGGGGGGGG", 10)
+    >>> builder.to_file(path = pathlib.Path("test.fasta"))
+Writing a FASTA with one contig with 100 A's and 50 T's
+    >>> from fgpyo.fasta.builder import FastaBuilder
+    >>> builder = FastaBuilder()
+    >>> builder.add("chr10").add("AAAAAAAAAA", 10).add("TTTTTTTTTT", 5)
+    >>> builder.to_file(path = pathlib.Path("test.fasta"))
+Add bases to existing contig
+    >>> from fgpyo.fasta.builder import FastaBuilder
+    >>> builder = FastaBuilder()
+    >>> contig_one = builder.add("chr10").add("AAAAAAAAAA", 1)
+    >>> contig_one.add("NNN", 1)
+    >>> contig_one.bases
+    'AAAAAAAAAANNN'
+"""
+import textwrap
+from pathlib import Path
+from typing import TYPE_CHECKING
+from typing import Any
+from typing import Dict
+
+""" Stubs for pysam imports """
+if TYPE_CHECKING:
+
+    def dict(*args: Any) -> None:
+        pass
+
+
+else:
+    from pysam import dict
+
+
+def pysam_dict(assembly: str, species: str, output_path: str, input_path: str) -> None:
+    """Calls pysam.dict and writes the sequence dictionary to the provided output path
+
+    Args
+        assembly: Assembly
+        species: Species
+        output_path: File path to write dictionary to
+        input_path: Path to fasta file
+    """
+    dict("-a", assembly, "-s", species, "-o", output_path, input_path)
+
+
+if TYPE_CHECKING:
+
+    def faidx(*args: Any) -> None:
+        pass
+
+
+else:
+    from pysam import faidx
```

Comment:
Same here
```suggestion
if TYPE_CHECKING:
    def samtools_faidx(*args: Any) -> None:
        pass
else:
    from pysam import faidx as samtools_faidx
```

Reply:
I'd perhaps also combine this and the `dict` one into a single if/else up above.